pypeit/arcimage.py

```diff
@@ -82,12 +87,18 @@ def save(self, outfile=None, overwrite=True):
             overwrite (:obj:`bool`, optional):
                 Overwrite any existing file.
         """
-        super(ArcImage, self).save(self.image, 'ARC', outfile=outfile, overwrite=overwrite,
-                                   raw_files=self.file_list, steps=self.process_steps)
-
-    # TODO: it would be better to have this instantiate the full class
-    # as a classmethod.
-    def load(self, ifile=None, return_header=False):
+        _outfile = self.master_file_path if outfile is None else outfile
+        # Check if it exists
+        if os.path.exists(_outfile) and not overwrite:
+            msgs.warn('Master file exists: {0}'.format(_outfile) + msgs.newline()
+                      + 'Set overwrite=True to overwrite it.')
+            return
+        #
+        hdr = self.build_master_header(steps=self.process_steps, raw_files=self.file_list)
+        pypeitimage.save_images(self.pypeitImage, _outfile, hdr=hdr, iext='ARC')
```

Comment:
Only because I had/have this philosophy to keep this object super light-weight.
Almost a container and nothing further.

I'm won't be adamant about it.  What are the pros/cons here?

Reply:
Having stewed on it a bit further and reminding myself that the `ImageMask` code is already inherited, I am going to push these into the class too.  Will still try to keep it quite light-weight but we'll see if we can toe that line.